orttraining/orttraining/python/checkpointing_utils.py

```diff
@@ -1,143 +1,124 @@
-    
+
     assert len(ckpt_file_names) > 0, "No checkpoint files found with prefix \"{}\" in directory {}.".format(checkpoint_prefix, checkpoint_dir)
     return ckpt_file_names
 
+
 def get_checkpoint_name(prefix, is_partitioned, world_rank=None, world_size=None):
-    SINGLE_CHECKPOINT_FILENAME='{prefix}.ort.pt'
-    MULTIPLE_CHECKPOINT_FILENAME='{prefix}.ZeRO.{world_rank}.{world_size}.ort.pt'
-    
+    SINGLE_CHECKPOINT_FILENAME = '{prefix}.ort.pt'
+    MULTIPLE_CHECKPOINT_FILENAME = '{prefix}.ZeRO.{world_rank}.{world_size}.ort.pt'
+
     if is_partitioned:
-        filename=MULTIPLE_CHECKPOINT_FILENAME.format(prefix=prefix, world_rank=world_rank, world_size=(world_size-1))
+        filename = MULTIPLE_CHECKPOINT_FILENAME.format(prefix=prefix, world_rank=world_rank, world_size=(world_size-1))
     else:
-        filename=SINGLE_CHECKPOINT_FILENAME.format(prefix=prefix)
-    
+        filename = SINGLE_CHECKPOINT_FILENAME.format(prefix=prefix)
+
     return filename
 
-class CombineZeroCheckpoint(object):
-    def __init__(self, checkpoint_files, clean_state_dict = None):
 
-        assert len(checkpoint_files) > 0, "No checkpoint files passed."
+def split_state_dict(state_dict):
+    optimizer_keys = ['Moment_1_', 'Moment_2_', 'Update_Count_', 'Step_']
+    split_sd = {'optimizer': {}, 'fp32_param': {}, 'fp16_param': {}}
+    for k, v in state_dict.items():
+        mode = 'fp32_param'
+        for optim_key in optimizer_keys:
+            if k.startswith(optim_key):
+                mode = 'optimizer'
+                break
+        if k.endswith('_fp16'):
+            mode = 'fp16_param'
+        split_sd[mode][k] = v
+    return split_sd
+
+
+class CombineZeroCheckpoint(object):
+    def __init__(self, checkpoint_files, clean_state_dict=None):
 
+        assert len(checkpoint_files) > 0, "No checkpoint files passed"
         self.checkpoint_files = checkpoint_files
         self.clean_state_dict = clean_state_dict
-        self.world_size = int(self.checkpoint_files[0].split('ZeRO')[1].split('.')[2]) +1
-        print(f"World size = {self.world_size}, expecting {self.world_size} files.")        
-        assert len(self.checkpoint_files) == self.world_size, "Could not find {} files".format(self.world_size)
-        
+        self.world_size = int(self.checkpoint_files[0].split('ZeRO')[1].split('.')[2]) + 1
+        assert len(self.checkpoint_files) == self.world_size, f"Could not find {self.world_size} files"
         self.weight_shape_map = dict()
-    
-    def _is_sharded(self, name):
-        if '_view_' in name:
-            return True
-        return False
-
-    def _has_fp16_weights(self, state_dict):
-        for k in state_dict.keys():
-            if k.endswith('_fp16'):
-                return True
-        return False
-    
-    def _split_moment_name(self, name):
+        self.sharded_params = set()
+
+    def _split_name(self, name):
         name_split = name.split('_view_')
```

Comment:
This PR completely shards the FP32 weight and adds the same _view_ suffix so that the sharded weight can be identified as such

Reply:
If fp32 state names also have ```_view``` in their names, it would break backwards compatibility. Which means, if there are users out there relying on the fp32 weight keynames not having ```_view_```, this change would break their script/code. Do we have any mitigation plan for that?

Reply:
This is a breaking change, and since we have experimental support for checkpointing, we do not promise backward compatibility. The user can use an earlier commit if required, although the preferred path would be to migrate to the newer usage.